src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,74 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_OF_FIGURES = 5;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure get() {
+        int randomGeneratedIndex = new Random().nextInt(NUMBER_OF_FIGURES);
+        switch (randomGeneratedIndex) {
+            case 0:
+                return getRectangle();
+            case 1:
+                return getSquare();
+            case 2:
+                return getIsoscelesTrapezoid();
+            case 3:
+                return getCircle();
+            case 4:
+            default:
+                return getRightTriangle();
+        }
+    }
+
+    private Rectangle getRectangle() {
+        Rectangle rectangle = new Rectangle();
+        rectangle.setRectangle(new Random().nextDouble(),
+                new Random().nextDouble(), new Random().nextDouble());
```

Comment:
let's not create new Random() each time

check all code in this class please

Reply:
add